packages/astro/src/core/build/plugins/plugin-middleware.ts

```diff
@@ -41,6 +41,17 @@ export function vitePluginMiddleware(
 				return 'export const onRequest = undefined';
 			}
 		},
+
+		writeBundle(_, bundle) {
+			for (const [chunkName, chunk] of Object.entries(bundle)) {
+				if (chunk.type === 'asset') {
+					continue;
+				}
+				if (chunk.fileName === 'middleware.mjs') {
+					internals.middlewareEntryPoint = new URL(chunkName, opts.settings.config.outDir);
```

Comment:
Should we combine `outDir` + `"server"` + `chunkName` here so we don't have to handle constructing the final `middlewareEntryPoint` again in`core/build/index.ts`? Otherwise this URL path would also point to a non existent path.

Reply:
Technically, at this point of the execution, the path **is correct**. The files are moved **later**.  If we decided to prefix `"server"` here, we would create a `URL` that point to a file that doesn't exist. If, in the future, we decide to expose this information to a hook that is called before the files are moved, we could incur an error.

Reply:
That's fair 👍 I'm mostly concerned that the final constructed path takes a lot of work to do so, which might not be great for perf. I won't let this block the PR though, I'm also fine with this for now.